src/components/ThreejsEditorModal.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import * as THREE from "three";
+import {ModalBody} from "react-bootstrap";
+
+import {ShowIf} from "./ShowIf";
+import settings from "../settings";
+import ModalDialog from "./ModalDialog";
+import {materialsToThreeDSceneData} from "../utils";
+import {LoadingIndicator} from "./LoadingIndicator";
+
+import {THREE_D_BASE_URL, THREE_D_SOURCES} from "../enums";
+
+export class ThreejsEditorModal extends ModalDialog {
```

Comment:
Let's isolate the logic for loading Three scripts to a separate barebones component and then reuse it in the Modal.

Reply:
It is already like that. All the functionality in here is about loading the threejs editor into dom.